code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -18,8 +18,20 @@
 	hunger_enabled = 0//Until automated eating mechanics are enabled, disable hunger for hostile mobs
 	var/shuttletarget = null
 	var/enroute = 0
+
+	// Vars to help find targets
 	var/list/targets = list()
 	var/attacked_times = 0
+	var/list/target_type_validator_map = list()
+	var/lowest_health = INFINITY // Max you can get
+	var/atom/T = null
+
+/mob/living/simple_animal/hostile/Initialize()
+	. = ..()
+	target_type_validator_map[/mob/living] = CALLBACK(src, .proc/validator_living)
+	target_type_validator_map[/obj/mecha] = CALLBACK(src, .proc/validator_mecha)
+	target_type_validator_map[/obj/machinery/bot] = CALLBACK(src, .proc/validator_bot)
+	return .

```

Comment:
okay style is one thing but `return .` is redundant once `.` has been initialized.